x/ccv/provider/types/keys.go

```diff
@@ -281,6 +296,27 @@ func getKeyPrefixes() map[string]byte {
 		// minimum power required to be in the top N per consumer chain.
 		MinimumPowerInTopNKeyName: 40,
 
+		// ConsumerIdKeyName is the key for storing the consumer id for the next registered consumer chain
+		ConsumerIdKeyName: 41,
+
+		// ConsumerIdToChainIdKeyName is the key for storing the chain id for the given consumer id
+		ConsumerIdToChainIdKeyName: 42,
+
+		// ConsumerIdToRegistrationRecordKeyName is the key for storing the registration record for the given consumer id
+		ConsumerIdToRegistrationRecordKeyName: 43,
+
+		// ConsumerIdToInitializationRecordKeyName is the key for storing the initialization record for the given consumer id
+		ConsumerIdToInitializationRecordKeyName: 44,
+
+		// ConsumerIdToOwnerAddressKeyName is the key for storing the owner address for the given consumer id
+		ConsumerIdToOwnerAddressKeyName: 45,
+
+		// ConsumerIdToPhaseKeyName is the key for storing the phase of a consumer chain with the given consumer id
+		ConsumerIdToPhaseKeyName: 46,
```

Comment:
Do we actually need this?

Reply:
I believe we do because when a chain stops, we should keep the registration and initialization record. If we do not keep it, I guess we can get rid of this state.
The reason, we need the phase state if we keep the records of a stopped chain is to distinguish between a chain that just registered & initialized, with a chain that registered & initialized & launched & stopped.

It also seems to me that it simplifies the code because now we can see exactly where a chain moves from a phase `a` to another phase `b`.